pkgs/top-level/all-packages.nix

```diff
@@ -4703,7 +4703,13 @@ in
     profiledCompiler = false;
   });
 
-  ghdl_mcode = callPackage ../development/compilers/ghdl { };
+  ghdl_mcode = callPackage ../development/compilers/ghdl {
+    flavour = "mcode";
```

Comment:
If you add `stdenv = stdenv_32bit;` line here, you should be able to build the package on x86_64 linux.